File: TypeScript/__tests__/algorithms/sort/Sort.test.ts
import { generateArrayWithRandomValues, isArraySorted } from '../../../src/util'

import { selectionSort } from '../../../src/algorithms/sorting/selectionSort'
import { combSort } from '../../../src/algorithms/sorting/combSort'

const array = generateArrayWithRandomValues(10000)

test('The selection sort algorithm must sort the array', () => {
  expect(isArraySorted(selectionSort(array))).toBe(true)
})

test('The comb sort algorithm must sort the array', () => {
  expect(isArraySorted(combSort(array))).toBe(true)
})

File: Go/tests/algorithms/sort_test.go
package sort_test

import (
	sort "Go/algorithms/sort/mergeSort"
	"Go/util"
	"testing"
)

func TestMergeSort(t *testing.T) {

	n := 10000
	max := n

	arr := util.GenerateArrayWithRandomValues(n, max)

	sort.MergeSort(arr, 0, len(arr))

	if !util.IsArraySorted(arr) {
		t.Error("must be true")
	}
}
